standalone-metastore/metastore-server/src/main/java/org/apache/hadoop/hive/metastore/utils/MetaStoreServerUtils.java

```diff
@@ -511,12 +513,12 @@ public static void clearQuickStats(Map<String, String> params) {
     params.remove(StatsSetupConst.NUM_ERASURE_CODED_FILES);
   }
 
-  public static void updateTableStatsForCreateTable(Warehouse wh, Database db, Table tbl,
+  public static boolean updateTableStatsForCreateTable(Warehouse wh, Database db, Table tbl,
```

Comment:
Maybe I don't get the context but why an update operations needs to return as a boolean? If something bad happens inside the update a dedicated exception should warn the caller, and can handle it. The return boolean is bit C++ style error management. Sometime could be OK, but I don't see the caller here.

Reply:
If it is only for unit test, you need to find out a another way, you are changing the return type of a public method, I don't think it is a compatible change.

It is functionally compatible, say anyone calling updateTableStatsForCreateTable() expecting a void will now get a boolean but he can ignore.

But this change isn't `Binary Compatible`, if the call is across Jars, at binary level it would be like removed a method with void and added another method with boolean as return type. So, those callers will fail with NoSuchMethod error, and for a UT it isn' required

Put some log or something like that and grep that in the tests...

Reply:
Fixed by using logging appender to capture and check presence of logging messages